server/src/controllers/order.controller.ts

```diff
@@ -0,0 +1,115 @@
+import { Request, Response } from 'express';
+import { Order } from '../models/index';
+import { createOrderSchema } from '../schema';
+import mongoose from 'mongoose';
+import { z } from 'zod';
+
+// Create a new order
+export const createOrder = async (req: Request, res: Response) => {
+    try {
+        // Validate request body against the schema
+        createOrderSchema.parse(req.body);
+
+        const newOrder = new Order(req.body);
+        const savedOrder = await newOrder.save();
+        return res.status(201).json({ data: savedOrder });
+    } catch (error) {
+        if (error instanceof z.ZodError) {
+            return res.status(400).json({ error: error.errors });
+        }
+        return res.status(400).json({ error: error instanceof Error ? error.message : 'An unexpected error occurred' });
+    }
+};
+
+// Get all orders
+export const getAllOrders = async (req: Request, res: Response) => {
+    const { page = 1, limit = 10 } = req.query;
+
```

Comment:
1. check if the decoded user is admin. yes->2. no->3.
2. check if the request comes with a query param (/orders?userId=blablabla), yes->4. no->5.
3. return all orders matching the decoded userId (the current logged in user)
4. return all orders matching the provided userId
5. return all orders in the db